source/i2c_api.c

```diff
@@ -321,4 +322,185 @@ void i2c_reset(i2c_t *obj)
     }
 }
 
+#if DEVICE_I2CSLAVE
```

Comment:
The API is not yet defined, thus I don't see why it should stay here. You can keep it on a branch for reference.